docs/pages/_app.js

```diff
@@ -215,6 +215,11 @@ function AppWrapper(props) {
     return { activePage, activePageParents, pages };
   }, [router.pathname]);
 
+  // Replicate change reverted in https://github.com/mui/material-ui/pull/35969/files#r1089572951
+  // Seems pretty safe to do in `mui-x` as all our pages are registered as paths.
+  // Fixes playground styles in dark mode.
+  const ThemeWrapper = pageContextValue.activePage ? ThemeProvider : React.Fragment;
```

Comment:
> Seems pretty safe to do in `mui-x` as all our pages are registered as paths.

While this might be the case right now, it might change in the future.
What if we detect playground pages by `pathname` instead?

```suggestion
  const ThemeWrapper = router.pathname.startsWith('/playground') ? React.Fragment : ThemeProvider;
```

Reply:
Great observation! Thanks for the safer suggestion. 😉